src/main/java/uk/gov/companieshouse/api/accounts/service/impl/SecretaryServiceImpl.java

```diff
@@ -0,0 +1,181 @@
+
+        SecretaryEntity entity = transformer.transform(rest);
+        entity.setId(secretaryId);
+
+        try {
+
+            secretaryRepository.save(entity);
+        } catch (MongoException e) {
+
+            throw new DataException(e);
+        }
+        return new ResponseObject<>(ResponseStatus.UPDATED, rest);
+    }
+
+    @Override
+    public ResponseObject<Secretary> find(String companyAccountsId, HttpServletRequest request) throws DataException {
+
+        SecretaryEntity entity;
+
+        try {
+
+            entity = secretaryRepository.findById(getSecretaryId(request)).orElse(null);
+
+        } catch (MongoException e) {
+
+            throw new DataException(e);
+        }
+
+        if (entity == null) {
+            return new ResponseObject<>(ResponseStatus.NOT_FOUND);
+        }
+
+        return new ResponseObject<>(ResponseStatus.FOUND, transformer.transform(entity));
+    }
+
+    @Override
+    public ResponseObject<Secretary> delete(String companyAccountsId, HttpServletRequest request) throws DataException {
+
+        String secretaryId = getSecretaryId(request);
+
+        try {
+            if (secretaryRepository.existsById(secretaryId)) {
+
+                secretaryRepository.deleteById(secretaryId);
+
+                directorsReportService
+                        .removeLink(companyAccountsId, DirectorsReportLinkType.SECRETARY, request);
+                return new ResponseObject<>(ResponseStatus.UPDATED);
+            } else {
+
+                return new ResponseObject<>(ResponseStatus.NOT_FOUND);
+            }
+        } catch (MongoException e) {
+
+            throw new DataException(e);
+        }
+    }
+
+    private void setMetadataOnRestObject(Secretary rest, Transaction transaction, String companyAccountsId, String secretaryId) {
+
+        rest.setLinks(createLinks(transaction, companyAccountsId, secretaryId));
+        rest.setEtag(GenerateEtagUtil.generateEtag());
+        rest.setKind(Kind.DIRECTORS_REPORT_SECRETARY.getValue());
+    }
+
+    private String getSelfLink(Secretary secretary) {
+
+        return secretary.getLinks().get(BasicLinkType.SELF.getLink());
+    }
+
+    private String getSecretaryId(HttpServletRequest request) {
+
+        String secretaryId = null;
+
+        Matcher matcher = SECRETARY_ID_REGEX.matcher(request.getRequestURI());
+        if (matcher.find()) {
+            secretaryId = matcher.group(1);
+        }
+
+        return secretaryId;
+
+    }
+    private String generateSelfLink(Transaction transaction, String companyAccountId, String secretaryId) {
+
+        return transaction.getLinks().getSelf() + "/"
+                + ResourceName.COMPANY_ACCOUNT.getName() + "/" + companyAccountId + "/"
+                + ResourceName.SMALL_FULL.getName() + "/"
+                + ResourceName.DIRECTORS_REPORT.getName() + "/"
+                + ResourceName.SECRETARIES.getName() + "/"
+                + secretaryId;
```

Comment:
secretaryId won't exist, so this can be removes and the `SECRETARIES` resource name renamed to `SECRETARY`